woodwork/tests/schema/test_schema_init.py

```diff
@@ -355,3 +355,27 @@ def test_schema_init_with_column_metadata(sample_column_names, sample_inferred_l
     schema = Schema(sample_column_names, sample_inferred_logical_types, column_metadata=column_metadata)
     for name, column in schema.columns.items():
         assert column['metadata'] == (column_metadata.get(name) or {})
+
+
+def test_no_schema_validation_valid_input(sample_column_names, sample_inferred_logical_types):
+    column_metadata = {
+        'age': {'interesting_values': [33]},
+        'signup_date': {'description': 'date of account creation'}
+    }
+    validated_schema = Schema(sample_column_names, sample_inferred_logical_types,
+                              column_metadata=column_metadata, index='signup_date', validate=True)
+    not_validated_schema = Schema(sample_column_names, sample_inferred_logical_types,
+                                  column_metadata=column_metadata, index='signup_date', validate=False)
+
+    assert validated_schema == not_validated_schema
+
+
+def test_no_schema_validation_invalid_input(sample_column_names, sample_inferred_logical_types):
+    error = 'semantic_tags for id must be a string, set or list'
+    with pytest.raises(TypeError, match=error):
+        Schema(sample_column_names, sample_inferred_logical_types,
+               semantic_tags={'id': 4}, validate=True)
+    error = "'int' object has no attribute 'union'"
+    with pytest.raises(AttributeError, match=error):
+        Schema(sample_column_names, sample_inferred_logical_types,
+               semantic_tags={'id': 4}, validate=False)
```

Comment:
I'm wondering if there is a better way to test these invalid input conditions. We don't really care so much about what error/message gets raised here...we only care that it is not coming from our Woodwork validation code. If the Python message were changed, this test would break, but everything would still be ok from our perspective.

One idea would be to use mocks for our validation functions (where possible) and use the mocks to check that our validation functions never get called instead of checking that some other non-Woodwork error is getting raised. Might want to think through this a little more to make sure we have a robust test.

Reply:
Totally agree. Was also feeling like it was a bit of a cop out to just check that one or two validation calls weren't getting called. Would be great to get something with better coverage. 

I need to read up more on mocks to understand what's possible, but I wonder if we could define a list of our validation functions (and make sure that validation is only ever happening with functions and not just inline) and confirm none of them get called with various input parameters being used.

Ideally we'd also have some way of raising an error if some new validation is added but it's not under validation control , we raise an error. I'll spend some time thinking about it.

Reply:
Here's one option of how we could test whether specific validation functions are getting called:

```python
@patch("woodwork.schema._validate_params")
def test_validation_method_called(mock_validation, sample_column_names, sample_inferred_logical_types):
    assert not mock_validation.called

    not_validated_schema = Schema(sample_column_names, sample_inferred_logical_types, validate=False)
    assert not mock_validation.called

    validated_schema = Schema(sample_column_names, sample_inferred_logical_types, validate=True)
    assert mock_validation.called

    assert validated_schema == not_validated_schema
```

I think if we add this kind of test for every validation function that's getting skipped when `validate=False`, that should make for good coverage.

Reply:
This approach seems solid to me and will allow us to verify that the validate parameter controls the function calls as we expect.